content/en/providers/cloud-compute/openstack/aai.md

```diff
@@ -470,6 +470,109 @@ members of `fedcloud.egi.eu`:
 ]
 ```
 
+## VO cleaning
+
+Sometimes it is easy to leave behind Virtual Machines that are no longer used,
+consuming unncessary resources. Owners of unused VMs should be notified to
+check whether occupied resources can be freed.
+
+EGI Check-in users get an `ePUID` (i.e. a long hash ending in `@egi.eu`) which
+are translated into local OpenStack user IDs. When VMs are created the owner of
+the VM is set to the OpenStack user ID instead of the `ePUID`. However, only
+the `ePUID` is linked to the user email in order for the user to be notified.
+The mapping between OpenStack user IDs and `ePUIDs` is shown with:
+
+```bash
+openstack user list
+```
+
+Problem is that regular users will not have the permissions to execute the
+command above. Below there are the steps to enable permissions only to staff
+of the EGI Foundation to execute the command, using the default keystone
+policy:
+
+```json
+ "identity:list_users": "(role:reader and system_scope:all) or (role:reader and domain_id:%(target.domain_id)s)"
+```
+
+### Step 1. Change the name of the egi.eu domain
```

Comment:
Not strictly needed

Reply:
If we don't set the name of the domain to egi.eu then it's set to domain ID auto-generated when creating the egi.eu identity provider so the other commands with --domain egi.eu won't work (including the user list command). One possibility could be to create the egi.eu domain with correct name before creating the egi.eu identity provider and pass it as parameter to "openstack identity provider create ..." command.

Reply:
Thanks @astalosj 
Would this be correct now?
https://docs.egi.eu/documentation/628/providers/cloud-compute/openstack/aai/#keystone-federation-support
![image](https://github.com/EGI-Federation/documentation/assets/5345517/678b3dfe-4cfd-47ac-8379-ca679d7cff9d)